client/a8c-for-agencies/sections/signup/controller.tsx

```diff
@@ -1,20 +1,46 @@
-import { type Callback } from '@automattic/calypso-router';
+import { Callback, Context } from '@automattic/calypso-router';
+import debug from 'debug';
+import store from 'store';
+import {
+	A4A_SIGNUP_FINISH_LINK,
+	A4A_SIGNUP_LINK,
+} from 'calypso/a8c-for-agencies/components/sidebar-menu/lib/constants';
 import { hideMasterbar } from 'calypso/state/ui/actions';
-import { getSignupDataFromLocalStorage } from './lib/signup-data-to-local-storage';
 import AgencySignUp from './primary/agency-signup';
+import AgencySignupFinish from './primary/agency-signup-finish';
 
 export const signUpContext: Callback = ( context, next ) => {
 	context.store.dispatch( hideMasterbar() );
 	context.primary = <AgencySignUp />;
 	next();
 };
 
-export const signUpResumeContext: Callback = ( context, next ) => {
-	const savedSignupData = JSON.stringify( getSignupDataFromLocalStorage() );
-	context.primary = (
-		<>
-			<div>{ savedSignupData }</div>
-		</>
-	);
+export const finishSignUpContext: Callback = ( context, next ) => {
+	context.store.dispatch( hideMasterbar() );
+	context.primary = <AgencySignupFinish />;
 	next();
 };
+
+type OverriddenPageContext = Context & { hash?: Record< string, string > };
+
+export const tokenRedirect: Callback = ( ctx, next ) => {
```

Comment:
We use similar logic as with `/connect` endpoint here to fill the tocken and then proceed to finish.